### PR TITLE
Test helper methods implementation

### PR DESCRIPTION
**Pull request by Mirageinvo**

### CODE REVIEW

**Comment by Mirageinvo** [comment]
@includealex PTAL

File: runtime/decoder.cpp [hunk 1]
```diff
@@ -19,10 +19,12 @@ bool Decoder::Destroy(Decoder *decoder) {
...
 Instruction Decoder::Decode(VMInstr operation) {
-    Opcode opc = static_cast<Opcode>(operation >> 24);
-    RegId_t r = (operation >> 16) & ~(TwoPow<16>() - TwoPow<8>());
-    Immediate_t imm = operation & ((1 << 16) - 1);
+    Opcode opc = static_cast<Opcode>(operation & 7);
+    RegId_t r = ((operation >> 8) & 7);
+    Immediate_t imm = 0;
+    uint32_t upper_value = ((operation >> 16) & 7);
+    uint32_t lower_value = (operation >> 24);
+    imm += (upper_value << 8) + lower_value;
```
Comment (includealex) [inline runtime/decoder.cpp:27]: We have no need of 24 str.
```c
Immediate_t imm = (upper_value << 8) + lower value;
```

**Comment by includealex** [comment]
@Mirageinvo  maybe it would be better to rename `instr_creator-inl.hpp` to `instr_creator_inl.hpp` and move it from `tests` folder to `includes`?

**Comment by Mirageinvo** [comment]
> @Mirageinvo maybe it would be better to rename `instr_creator-inl.hpp` to `instr_creator_inl.hpp` and move it from `tests` folder to `includes`?

I suppose, we need instr_creator-inl.hpp only for testing purposes since we will have our instructions generated by frontend. So, I suggest to leave it as it is.

**Comment by includealex** [comment]
After conversation, we've decided to remove it.

File: runtime/decoder.cpp (hunk 1 above)
Comment (Mirageinvo) [reply runtime/decoder.cpp]: Done

**Review by includealex — APPROVED** [review]
@Mirageinvo now it looks good for me.

**Pull request merged**